site/docs/Guides-ConstructingTypes.md

```diff
@@ -0,0 +1,116 @@
+---
+title: Constructing Types
+layout: ../_core/DocsLayout
+category: Guides
+permalink: /graphql-js/constructing-types/
+---
+
+For many apps, you can define a fixed schema when the application starts, and define it using GraphQL schema language. In some cases, it's useful to construct a schema programmatically. You can do this using the `GraphQLSchema` constructor.
+
+When you are using the `GraphQLSchema` constructor to create a schema, instead of defining `Query` and `Mutation` types solely using schema language, you create them as separate object types using the `GraphQLObjectType` constructor.
+
+For example, let's say we are building a simple API that lets you fetch user data for a few hardcoded users based on an id. Using `buildSchema` we could write a server with:
+
+```javascript
+var express = require('express');
+var graphqlHTTP = require('express-graphql');
+var { buildSchema } = require('graphql');
+
+var schema = buildSchema(`
+  type User {
+    id: String
+    name: String
+  }
+
+  type Query {
+    user(id: String): User
+  }
+`);
+
+// Maps id to User object
+var fakeDatabase = {
+  'a': {
+    id: 'a',
+    name: 'alice',
+  },
+  'b': {
+    id: 'b',
+    name: 'bob',
+  },
+};
+
+var root = {
```

Comment:
We chatted about this over the phone but TLDR this example comes later under "guides" so I think the ordering is clear